### PR TITLE
Networking Functions for Flight Computer and Possible Control loop structure

### PR DESCRIPTION
**Pull request by lwaghorn**
### Known Bug
When I place the functions in NodeServer.cpp inside of PodInternalNetwork.cpp the server doesn't work anymore, which makes 0 sense to me. But in the meantime its separate. 

### Current Functionality
Right now main creates two sockets, one for the client and one for the server. Then it spawns a thread for the client and server passing them their respective sockets. This will most likely be replaced with a pointer to a pod object.

The client makes proto packets as described by the ProtoStructs.proto file, serializes them and sends it over the client socket. The server socket receives this, parses the message back into a proto object and that's it. There's a function there to parse it but nothing is hooked up yet.

### Next Steps
I'm going to create some sort of base class that will manage the getter/setter methods. Allowing us to specify the permissions for each thread in the class. Then work on the shared memory aspect between threads

### Biggest Concerns
The maincontrolloop.cpp. What's in there is just a rough idea on how to do it, it's not complete.  To be honest I'm not a big fan of it. Critical review on that would be wicked. We can scrap it if need be.

### CODE REVIEW

File: OnPod/FlightComputer/include/FlightComputer/structs.h [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  eTerminalStates TerminalState;
+  eMotorStates MotorState;
+  eBreakNodeStates BreakNodeState;
+  // Navigation
+  float fDistance;
+  float fVelocity;
+  // Rear Node
+  float fGpioValues;
+  //FlagsV2
+  char cFlagsArray[3] = {0};
```
Comment (Newfloridadad) [inline OnPod/FlightComputer/include/FlightComputer/structs.h]: unsigned char for flag buffers, similar with ints below, define if unsigned/signed and size of the value uint32_t

File: OnPod/FlightComputer/src/MainControlLoop.cpp [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  if ((terminalState == tcTerminalFlight) & (nodeState == bnsFlight))
+  {
+    Pod->setTerminalCommand(tcTerminalNone);
+    Pod->setMotorState(msDrive);
+    return true;
+  }
+  return false;
+}
+
+int runControlLoop(MemoryAccess Pod)
```
Comment (Newfloridadad) [inline OnPod/FlightComputer/src/MainControlLoop.cpp]: missing *?

File: OnPod/FlightComputer/src/MemoryAccess.cpp [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+
+
+ MemoryAccess::MemoryAccess(PodValues sPodValues)
+ {
+	 this->sPodValues = sPodValues;
+ };
+
+
+ int MemoryAccess::setPodState(ePodStates state)
```
Comment (Newfloridadad) [inline OnPod/FlightComputer/src/MemoryAccess.cpp]: is there a normal convention for returning ints not bools here?

File: OnPod/FlightComputer/src/NodeServer.cpp [hunk 4]
```diff
@@ -0,0 +1,125 @@
+#include "FlightComputer/PodInternalNetwork.h"
+#include "FlightComputer/MemoryAccess.h"
+#define UDPPORT 5008
+
+using namespace fc;
+
+void parseBreakNodePacket(fc::brakeNodeData pPayload, MemoryAccess Pod){
+	//Grab State
+	switch (pPayload.state()){
```
Comment (Newfloridadad) [inline OnPod/FlightComputer/src/NodeServer.cpp]: is this switch irrelevent since each time we'll parse the packet and store the state no matter the state?

**Review by Newfloridadad — COMMENTED** [review]
A few style comments/general questions. building now

File: OnPod/FlightComputer/src/NodeServer.cpp (hunk 4 above)
Comment (lwaghorn) [reply OnPod/FlightComputer/src/NodeServer.cpp]: ya you're right. Im gonna change the type in the pod values struct for PodState to use the enum type make by the protobuf file

Comment (lwaghorn) [reply OnPod/FlightComputer/src/NodeServer.cpp]: ya you're right. I'm gonna change the BrakeNodeState var in the pod values to use the enum type made by the protofile. 

Good catch

File: OnPod/FlightComputer/src/MainControlLoop.cpp (hunk 2 above)
Comment (lwaghorn) [reply OnPod/FlightComputer/src/MainControlLoop.cpp]: Pass the memoryAccess object by value. But it contains a pointer to the shared memory struct

**Pull request merged**